app/src/main/kotlin/at/bitfire/davdroid/settings/AccountSettings.kt

```diff
@@ -137,34 +135,21 @@ class AccountSettings @AssistedInject constructor(
 
 
     val accountManager: AccountManager = AccountManager.get(context)
-    val account: Account = when (accountOrAddressBookAccount.type) {
-            context.getString(R.string.account_type_address_book) -> {
-                /* argument is an address book account, which is not a main account. However settings are
-                stored in the main account, so resolve and use the main account instead. */
-                LocalAddressBook.mainAccount(context, accountOrAddressBookAccount) ?: throw IllegalArgumentException("Main account of $accountOrAddressBookAccount not found")
-            }
-
-            context.getString(R.string.account_type),
-            "at.bitfire.davdroid.test" /* defined in androidTest/strings/account_type_test */ ->
-                accountOrAddressBookAccount
-
-            else ->
-                throw IllegalArgumentException("Account type ${accountOrAddressBookAccount.type} not supported")
-        }
-
     init {
+        val requiredAccountType = context.getString(R.string.account_type)
+        if (account.type != requiredAccountType)
+            throw IllegalArgumentException("Invalid account type: ${account.type} (must be $requiredAccountType)")
```

Comment:
If we not throw this exception. Tests which use the test account type (at.bitfire.davdroid.test) all fail. I am not sure how to handle that? 

Rewriting the source to handle test cases is out of the question.

We could expect the illegal argument exception in tests, but then it becomes harder to test certain functionality.

Or we could drop the test account type and use the normal at.bitfire.davdroid type in testing too?

Reply:
> 3\. Or we could drop the test account type and use the normal at.bitfire.davdroid type in tests too?

We can't do that because when a normal account is created, the sync framework automatically starts background sync adapters which interfere with the tests. This is the main reason for the test account type.

> 1\. Rewriting the source to handle test cases is out of the question.

Maybe the formally most correct version would be to have an `AccountTypeProvider` which is injected over Hilt, and which provides a different value for tests.

However because the `AccountSettings` will soon be rewritten to use the DB instead of accounts, this would be useless work. And the one line doesn't hurt much until then …